src/pipecat/audio/vad/vad_analyzer.py

```diff
@@ -183,36 +183,37 @@ def analyze_audio(self, buffer) -> VADState:
         if len(self._vad_buffer) < num_required_bytes:
             return self._vad_state
 
-        audio_frames = self._vad_buffer[:num_required_bytes]
-        self._vad_buffer = self._vad_buffer[num_required_bytes:]
-
-        confidence = self.voice_confidence(audio_frames)
-
-        volume = self._get_smoothed_volume(audio_frames)
-        self._prev_volume = volume
-
-        speaking = confidence >= self._params.confidence and volume >= self._params.min_volume
-
-        if speaking:
-            match self._vad_state:
-                case VADState.QUIET:
-                    self._vad_state = VADState.STARTING
-                    self._vad_starting_count = 1
-                case VADState.STARTING:
-                    self._vad_starting_count += 1
-                case VADState.STOPPING:
-                    self._vad_state = VADState.SPEAKING
-                    self._vad_stopping_count = 0
-        else:
-            match self._vad_state:
-                case VADState.STARTING:
-                    self._vad_state = VADState.QUIET
-                    self._vad_starting_count = 0
-                case VADState.SPEAKING:
-                    self._vad_state = VADState.STOPPING
-                    self._vad_stopping_count = 1
-                case VADState.STOPPING:
-                    self._vad_stopping_count += 1
+        while len(self._vad_buffer) >= num_required_bytes:
```

Comment:
@markbackman, this was another needed change due to the new resampler. 

Since we can now receive audio chunks that are bigger, the `analyze_audio` method needs to process all available data. So I had to add this loop here to properly analyze the entire audio buffer in the VAD analyzer.